### ISSUE TITLE
What can be the reason behind getting same values for mean during prediction of multiclass classification in gpflow model

### ISSUE THREAD

**Issue by irum** [GPflow/GPflow#1292]
I am using the multiclass classification program given in documentation of gpflow for classification problem . I am using my own image data. when I execute the program i am getting values for inducing points in this format 3.99595623e-199 . Are these values fine ? The all values of mean are same i.e 0.09993924 when I use predict_y but when I use predict_f then it gives me different values which I am not understanding why? I have already asked this question on stack overflow but i have not got any reply yet. Please help me on this.
The code is as follows:

    from typing import Tuple
    import numpy as np
    import tensorflow as tf
    import gpflow
    from gpflow.ci_utils import ci_niter
    import tensorflow2_work.utils as utils
    from gpflow.config import default_float
    from gpflow.utilities import print_summary, set_trainable
    import warnings
    Dir = "/media/irum/DATA/Dataset/train"
    Dir_test = "/media/irum/DATA/Dataset/test"
     warnings.filterwarnings('ignore')

    tf.executing_eagerly()

    gpflow.config.set_default_float(np.float64)

    is_training=True
    C=10
    batch_size=32

    kernel = gpflow.kernels.Matern32() + gpflow.kernels.White(variance=0.01)

    invlink = gpflow.likelihoods.RobustMax(C)  # Robustmax inverse link function
    likelihood = gpflow.likelihoods.MultiClass(C, invlink=invlink)  # Multiclass likelihood

    inducing_variable = np.zeros((10,150528), default_float())

    model = gpflow.models.SVGP(kernel=kernel, likelihood=likelihood,
                inducing_variable=inducing_variable, num_latent=C, whiten=True, q_diag=True)

    set_trainable(model.kernel.kernels[1].variance,True)
    set_trainable(model,True)
    set_trainable(model.inducing_variable,True)

    optimizer=tf.optimizers.Adam(0.001)

    @tf.function
    def optimization_step(model: gpflow.models.SVGP, x,y):

        with tf.GradientTape(watch_accessed_variables=False) as tape:
             tape.watch(model.trainable_variables)
             obj = - model.log_marginal_likelihood(x,y)
             obj=obj/batch_size
             grads = tape.gradient(obj, model.trainable_variables)
       optimizer.apply_gradients(zip(grads,model.trainable_variables))

       return obj

     def simple_training_loop(model: gpflow.models.SVGP, epochs: int = 3, logging_epoch_freq: int = 
                      10):

                           data_loader = utils.Data_loader(32, Dir)
                            total_loss = []
                            total_acc = []  

                             for epoch in range(epochs):
       
                                         data_loader.shuffle_data(is_training)

                                         train_loss = 0.
                                          acc_loss = 0

                                           for _ in range(data_loader.n_batches):
                                                           batch_x,batch_y = data_loader.next_batch(32)
                                                           batch_x=tf.convert_to_tensor(batch_x)
                                                           batch_y=tf.convert_to_tensor(batch_y)
                                                           batch_x=tf.cast(batch_x, dtype=default_float())
                                                        #  batch_y=tf.cast(batch_y,dtype=default_float())
                                                           batch_x = tf.reshape(batch_x, [32, -1])

                                                           loss=optimization_step(model, batch_x,batch_y)
                                                          epoch_id = epoch + 1
                                                         if epoch_id % logging_epoch_freq == 0:
                                                         tf.print(f"Epoch {epoch_id}: ELBO (train)                                                         
                                                        {(model.log_marginal_likelihood(batch_x,batch_y))/batch_size}")

                
                                        m, v = model.predict_y(batch_x)
                                        print(model.inducing_variable.Z.numpy())
                                        print(model.kernel.trainable_variables)
                                        preds = np.argmax(m, 1).reshape(batch_y.shape)
                                        correct = (preds == batch_y.numpy().astype(int))
                                        acc = np.average(correct.astype(float)) * 100.

                                         train_loss+=loss
                                         acc_loss+=acc

                                         total_loss.append(loss.numpy())
                                          total_acc.append(acc)

    print('Accuracy is {}%'.format(np.mean(np.array(total_acc))))
    print('total loss is {}'.format(np.mean(np.array(total_loss))))

  
     
     if __name__ =="__main__" :
                   simple_training_loop(model, epochs=10, logging_epoch_freq=2)

**Comment by st--** [comment]
Thanks for your interest in GPflow! As we are just volunteering our spare time to this project, we cannot always respond to questions immediately, especially when they take a bit more time of digging into. As this question does not directly concern the code of GPflow, it is more appropriate to [StackOverflow, where it has already been posted](https://stackoverflow.com/questions/60425123/what-can-be-the-reason-behind-getting-same-values-for-mean-during-prediction-of).
